### PR TITLE
[UI v2] feat: scaffolds automations route

### PR DESCRIPTION
**Pull request by devinvillarosa**
1. Creates routes for `/automations`, `/automations/create`, and `/automations/automation/:id`
2. Adds automation empty state component
3. Adds automations header component

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: ui-v2/src/components/ui/docs-link.tsx [hunk 1]
```diff
@@ -17,13 +19,14 @@ type DocsID = keyof typeof DOCS_LINKS;
 
 type Props = {
 	id: DocsID;
+	label?: "View Docs" | "Documentation";
```
Comment (devinvillarosa) [inline ui-v2/src/components/ui/docs-link.tsx:22]: Need to make this update. Seems like docs link can either be `View Docs` or `Documentation`

File: ui-v2/src/components/ui/docs-link.tsx [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { Icon } from "./icons";
 import { Button } from "./button";
 
 const DOCS_LINKS = {
+	"automations-guide":
+		"https://docs.prefect.io/v3/automate/index#automate-overview",
```
Comment (desertaxle) [inline ui-v2/src/components/ui/docs-link.tsx:9]: I think we want this more specific page instead:
```suggestion
	"automations-guide": 	"https://docs.prefect.io/v3/automate/events/automations-triggers",
```

File: ui-v2/src/routes/automations/index.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { createFileRoute } from "@tanstack/react-router";
 
-export const Route = createFileRoute("/automations")({
+export const Route = createFileRoute("/automations/")({
```
Comment (desertaxle) [inline ui-v2/src/routes/automations/index.ts:3]: I'm pretty sure we don't want the trailing slash to match the other routes, but is there a reason the trailing slash is needed?
```suggestion
export const Route = createFileRoute("/automations")({
```

Comment (devinvillarosa) [reply ui-v2/src/routes/automations/index.ts:3]: I think its autogenerated from `tanstack router`, when you move it to a nested folder.
Not sure if thats intentional from the library

https://github.com/user-attachments/assets/76853980-7dd9-4832-be7d-523733048f9f

Comment (devinvillarosa) [reply ui-v2/src/routes/automations/index.ts:3]: https://github.com/user-attachments/assets/cbfb1da8-ea13-4d91-b5f0-d8f814ce1ef8

File: ui-v2/src/components/ui/docs-link.tsx (hunk 2 above)
Comment (devinvillarosa) [reply ui-v2/src/components/ui/docs-link.tsx]: thanks! updated in latest push

**Review by desertaxle — APPROVED** [review]

**Pull request merged**